sktime/forecasting/compose/_pipeline.py

```diff
@@ -1448,3 +1450,192 @@ def get_test_params(cls, parameter_set="default"):
+    including ForecastingPipeline and TransformedTargetForecaster.
+
+    The `steps_arg` parameter needs to be pointed to the "steps"-like parameter
+    of the wrapped forecaster and `permutation` switches the sequence of steps.
+
+    Not very useful on its own, but
+    useful in combination with tuning or auto-ML wrappers on top of this.
+
+    Parameters
+    ----------
+    estimator : sktime forecaster, inheriting from BaseForecaster
+        must have parameter with name `steps_arg`
+        estimator whose steps are being permuted
+    permutation : list of str, or None, optional, default = None
+        if not None, must be equal length as getattr(estimator, steps_arg)
+        and elements must be equal to names of estimator.steps_arg estimators
+        names are unique names as created by _get_estimator_tuples (if unnamed list),
+        or first string element of tuples, of estimator.steps_arg
+        list is interpreted as range of permutation of names
+        if None, is interpreted as the identity permutation
+    steps_arg : string, optional, default="steps"
+        name of the steps parameter. getattr(estimator, steps_arg) must be
+        list of estimators, or list of (str, estimator) pairs
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> from sktime.forecasting.compose import ForecastingPipeline, Permute
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.transformations.series.boxcox import BoxCoxTransformer
+    >>> from sktime.transformations.series.exponent import ExponentTransformer
+
+    Simple example: permute sequence of estimator in forecasting pipeline
+    >>> y = load_airline()
+    >>> fh = ForecastingHorizon([1, 2, 3])
+    >>> pipe = ForecastingPipeline(
+    ...     [
+    ...         ("boxcox", BoxCoxTransformer()),
+    ...         ("exp", ExponentTransformer(3)),
+    ...         ("naive", NaiveForecaster()),
+    ...     ]
+    ... )
+    >>> # this results in the pipeline with sequence "exp", "boxcox", "naive"
+    >>> permuted = Permute(pipe, ["exp", "boxcox", "naive"])
+    >>> pipe = pipe.fit(y, fh=fh)
+    >>> y_pred = pipe.predict()
+
+    The permuter is useful in combination with grid search (toy example):
+    >>> from sktime.datasets import load_shampoo_sales
+    >>> from sktime.forecasting.model_selection import (
+    ...     ExpandingWindowSplitter,
+    ...     ForecastingGridSearchCV,
+    ... )
+    >>> fh = [1,2,3]
+    >>> cv = ExpandingWindowSplitter(
+    ...     start_with_window=True,
+    ...     fh=fh)
+    >>> forecaster = NaiveForecaster()
+    >>> # check which of the two sequences of transformers is better
+    >>> param_grid = {
+    ...     "permutation" : [["boxcox", "exp", "naive"], ["exp", "boxcox", "naive"]]
+    ... }
+    >>> gscv = ForecastingGridSearchCV(
+    ...     forecaster=forecaster,
```

Comment:
I am somewhat confused by this example.  Shouldn't the "forecaster" here be `permuted` from above? Not NaiveForecaster()?

Reply:
it should be `permuted` indeed. Why does this not raise an error?

Reply:
fixed